File: src/main/java/org/aikidodanang/model/Post.java
package org.aikidodanang.model;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "posts")
public class Post {
    @Id
    private String id;
    @Field("title")
    private String title;
    @Field("content")
    private String content;
    @Field("userName")
    private String userName;
    @Field("heading")
    private String heading;
    @Field("subHeading")
    private String subHeading;
    @Field("article")
    private boolean article;
    @Field("createDate")
    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    @JsonSerialize(using = LocalDateTimeSerializer.class)
    private LocalDateTime createDate;
}

File: src/main/kotlin/org/aikidodanang/model/Post.kt
package org.aikidodanang.model

import org.aikidodanang.dto.PostDto
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime

@Document(collection = "posts")
data class Post(
        val id: String,
        val title: String,
        val image: String,
        val heading: String,
        val subHeading: String,
        val content: String,
        val createDate: LocalDateTime
) {
    companion object {
        fun fromPostDto(postDto: PostDto) = Post(
                id = postDto.id,
                title = postDto.title,
                image = postDto.image,
                heading = postDto.heading,
                subHeading = postDto.subHeading,
                content = postDto.content,
                createDate = postDto.createDate
        )
    }
}